caom2-artifact-resolvers/src/test/java/ca/nrc/cadc/caom2/artifact/resolvers/MastResolverTest.java

```diff
@@ -121,8 +121,11 @@ public void testValidURI() {
 
                 URI uri = new URI(uriStr);
                 URL url = mastResolver.toURL(uri);
+                String schemeSpecificPart = uri.getSchemeSpecificPart();
 
-                Assert.assertTrue(url.getPath().endsWith(uri.getSchemeSpecificPart()));
+                log.debug("toURL returned: " + url.toString());
+                Assert.assertTrue(url.toString().equals( MAST_BASE_URL + schemeSpecificPart));
+                Assert.assertTrue(url.getPath().endsWith(schemeSpecificPart));
             }
```

Comment:
This test is essentially doing the same work as the MastResolver so isn't proving anything.  You could simply look to make sure there's a slash before the scheme specific part of the URI.  We don't want the test relying on the baseURL of the MAST services.